spark3/src/main/java/org/apache/iceberg/spark/procedures/RewriteManifestsProcedure.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.procedures;
+
+import org.apache.iceberg.actions.Actions;
+import org.apache.iceberg.actions.RewriteManifestsActionResult;
+import org.apache.iceberg.spark.procedures.SparkProcedures.ProcedureBuilder;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+class RewriteManifestsProcedure extends BaseProcedure {
+
+  private static final ProcedureParameter[] PARAMETERS = new ProcedureParameter[]{
```

Comment:
I just realized that there are more configs we should probably expose as they are supported by the action. I think we can expose optional `spec_id`, `use_caching`. I am not convinced we want to expose `stagingLocation`, though. 

@rdblue @RussellSpitzer, thoughts?

Reply:
We will have more params to track min/max sizes, clustering ration later on. I don't want to have too many params.

Reply:
I think use_caching may be important, just because of the use case hitting us right now internally

Reply:
Agree, I added those two.

Reply:
I would avoid staging location for now because it defaults to the table's metadata location. I'm also not convinced that we should expose `spec_id` because it is an internal identifier that we never show to users right now. It makes sense in the API, but I doubt a purely SQL user would ever know what to set it to.

Reply:
Well, there is no easy way to find the spec id without using the Iceberg API directly so I agree we can skip it for now.

Reply:
Done.